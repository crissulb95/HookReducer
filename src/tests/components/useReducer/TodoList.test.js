const { shallow } = require("enzyme");
const { TodoList } = require("../../../components/useReducer/TodoList");
const { demos } = require("../../fixtures/demoTodos");

describe('Pruebas en <TodoList />', () => {

    const handleDelete = jest.fn();
    const handleTachado = jest.fn();
    
    const wrapper = shallow(
        <TodoList 
        todos = {demos}
        handleTachado={handleTachado}
        handleDelete={handleDelete}
        />
    );

    test('Debe mostrarse correctamente', () => {
        
        expect(wrapper).toMatchSnapshot();

    })
    
    test('Debe tener dos <TodoItem />', () => {
        
        expect(wrapper.find('TodoItem').length).toBe(2)
        expect(wrapper.find('TodoItem').length).toBe(demos.length)
        expect(wrapper.find('TodoItem').at(0).prop("handleTachado")).toEqual(expect.any(Function))
        
    })
    

})
