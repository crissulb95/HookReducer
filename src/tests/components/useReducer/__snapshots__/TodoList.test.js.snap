// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <TodoList /> Debe mostrarse correctamente 1`] = `
<ul
  className="list-group list-group-flush"
>
  <TodoItem
    handleDelete={[MockFunction]}
    handleTachado={[MockFunction]}
    index={0}
    key="1"
    todo={
      Object {
        "description": "testing description 1",
        "done": false,
        "id": 1,
      }
    }
  />
  <TodoItem
    handleDelete={[MockFunction]}
    handleTachado={[MockFunction]}
    index={1}
    key="2"
    todo={
      Object {
        "description": "testing description 2",
        "done": false,
        "id": 2,
      }
    }
  />
</ul>
`;
